src/WordPress/NumArgsAwareTrait.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Dhii\EventManager\WordPress;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+use OutOfRangeException;
+
+/**
+ * Common functionality for awareness of the number of arguments to be passed to native WordPress hook handlers.
+ *
+ * @since [*next-version*]
+ */
+trait NumArgsAwareTrait
+{
+    /**
+     * The argument count value to pass to the native WordPress action and filter hooking functions.
+     *
+     * @since [*next-version*]
+     *
+     * @var int
+     */
+    protected $numArgs;
+
+    /**
+     * Retrieves the argument count value to pass to the native WordPress action and filter hooking functions.
+     *
+     * @since [*next-version*]
+     *
+     * @return int The number of arguments.
+     */
+    protected function _getNumArgs()
+    {
+        return $this->numArgs;
```

Comment:
The property may have the value of `null`, because it can be un-initialized. Because this method promises an integer, I think it could be improved by casting the number to `(int)`.